src/seedu/addressbook/AddressBook.java

```diff
@@ -579,6 +588,47 @@ private static String executeListAllPersonsInAddressBook() {
         return getMessageForPersonsDisplayedSummary(toBeDisplayed);
     }
 
+    /**
+     * Displays all persons in the address book to the user; sorted by second command word
+     * @return feedback display message for the operation result
+     */
+    private static String executeSortAllPersonsInAddressBook(String arg) {
+        String trimmedArg = arg.trim();
+        int commandType = INVALID_COMMAND;
+        for (int i = 0; i < COMMAND_SORT_SUFFIXES.length; i++) {
+            if (trimmedArg.equals(COMMAND_SORT_SUFFIXES[i])) {
+                commandType = i;
+            }
+        }
+        if (commandType == INVALID_COMMAND) {
+            return getMessageForInvalidCommandInput(COMMAND_SORT_WORD, getUsageInfoForSortCommand());
+        }
+        ArrayList<String[]> toBeSorted = new ArrayList<>(getAllPersonsInAddressBook());
+        ArrayList<String[]> toBeDisplayed = new ArrayList<>();
+        String[] nextPerson;
+
+        while (!toBeSorted.isEmpty())
+        {
+            nextPerson = toBeSorted.get(0);
+            for (String[] person: toBeSorted)
+            {
+                if (isSecondPersonSmaller(nextPerson, person, commandType)) {
+                    nextPerson = person;
+                }
+            }
+            toBeDisplayed.add(nextPerson);
+            toBeSorted.remove((nextPerson));
+        }
+        showToUser(toBeDisplayed);
+        return getMessageForPersonsDisplayedSummary(toBeDisplayed);
+
+    }
```

Comment:
this entire method seems a bit too long, perhaps can increase the SLAP (Single level abstraction principle)